core/src/main/scala/org/apache/spark/api/java/JavaRDD.scala

```diff
@@ -99,6 +99,8 @@ class JavaRDD[T](val rdd: RDD[T])(implicit val classTag: ClassTag[T])
 
   /**
    * Return a sampled subset of this RDD.
+   * Note: this is NOT guaranteed to provide exactly the fraction of the count
```

Comment:
There are a couple overloads of sample here; update them all and maybe apply the clarification about seed you added below?